tests/python/test_serving_unit.py

```
@@ -126,3 +128,531 @@ def test_parse_no_prefix_cache(self):
+        from dashinfer.serving.protocol import ResponseFormat
+        from dashinfer.serving.server import _build_response_format
+        schema = {"type": "object", "properties": {"z": {"type": "number"}}}
+        rf = ResponseFormat(type="json_schema", json_schema=schema)
+        result = _build_response_format(rf)
+        assert json.loads(result["json_schema"]) == schema
+
+    def test_invalid_type_rejected_by_protocol(self):
+        """ResponseFormat.type is a Literal — Pydantic rejects unknown values."""
+        import pytest
+        from dashinfer.serving.protocol import ResponseFormat
+        with pytest.raises(Exception):  # ValidationError
```

Ruff (0.15.1)

[warning] Do not assert blind exception: `Exception`

(B017)